### PR TITLE
Redesign SetupWizard activities

### PR DESCRIPTION
**Pull request by chirayudesai**

### CODE REVIEW

**Comment by chirayudesai** [comment]
AOSP build works.

Gradle build fails to find `SudThemeGlifV3.Light`

File: app/src/main/java/com/stevesoltys/seedvault/restore/RestoreFilesFragment.kt [hunk 1]
```diff
@@ -26,12 +28,16 @@ internal class RestoreFilesFragment : SnapshotFragment() {
 
         val topStub: ViewStub = v.findViewById(R.id.topStub)
         topStub.layoutResource = R.layout.header_snapshots
-        topStub.inflate()
+        val header = topStub.inflate()
+        val suw_layout: GlifLayout = header.findViewById(R.id.setup_wizard_layout)
```
Comment (grote) [inline app/src/main/java/com/stevesoltys/seedvault/restore/RestoreFilesFragment.kt:32]: nitpick: variable names in Kotlin should use camelCase, no underscores.

edit: this also goes for all other places where this variable appears.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -148,8 +148,8 @@
     <string name="restore_title">Restore from backup</string>
     <string name="restore_choose_restore_set">Choose a backup to restore</string>
     <string name="restore_restore_set_times">Last backup %1$s · First %2$s.</string>
-    <string name="restore_back">Don\'t restore</string>
-    <string name="restore_skip">Skip restoring apps</string>
+    <string name="restore_skip">Don\'t restore</string>
+    <string name="restore_skip_apps">Skip restoring apps</string>
```
Comment (grote) [inline app/src/main/res/values/strings.xml:152]: This is likely to mess with translations. Maybe don't rename or at least chose completely new names?

File: gradle/dependencies.gradle [hunk 3]
```diff
@@ -21,6 +21,10 @@ ext.aosp_libs = fileTree(include: [
     'android.jar',
     // out/target/common/obj/JAVA_LIBRARIES/core-libart.com.android.art_intermediates/classes.jar
     'libcore.jar',
+    // out/target/product/device/obj/JAVA_LIBRARIES/setupcompat_intermediates/javalib.jar
+    'setupcompat.jar',
+    // out/target/product/device/obj/JAVA_LIBRARIES/setupdesign_intermediates/javalib.jar
+    'setupdesign.jar',
```
Comment (grote) [inline gradle/dependencies.gradle:27]: If those are not automatically generated, we should also document on how to get them.

File: app/src/main/java/com/stevesoltys/seedvault/restore/RestoreFilesFragment.kt [hunk 4]
```diff
@@ -54,6 +60,11 @@ internal class RestoreFilesStartedFragment : Fragment() {
     ): View {
         val v: View = inflater.inflate(R.layout.fragment_restore_files_started, container, false)
 
+        val suw_layout: GlifLayout = v.findViewById(R.id.setup_wizard_layout)
```
Comment (grote) [inline app/src/main/java/com/stevesoltys/seedvault/restore/RestoreFilesFragment.kt:63]: Maybe we can keep a reference to this, so we don't have to find it two times?

File: app/src/main/java/com/stevesoltys/seedvault/ui/RequireProvisioningActivity.kt [hunk 5]
```diff
@@ -32,6 +33,7 @@ abstract class RequireProvisioningActivity : BackupActivity() {
         }
     private val requestLocation =
         registerForActivityResult(StartActivityForResult()) { result ->
+            setResult(result.resultCode)
```
Comment (grote) [inline app/src/main/java/com/stevesoltys/seedvault/ui/RequireProvisioningActivity.kt:36]: Why do we need to set a result code here and above?

File: app/src/main/java/com/stevesoltys/seedvault/ui/recoverycode/RecoveryCodeInputFragment.kt [hunk 6]
```diff
@@ -105,8 +103,6 @@ class RecoveryCodeInputFragment : Fragment() {
 
         if (viewModel.isRestore) {
             introText.setText(R.string.recovery_code_input_intro)
-            backView.visibility = VISIBLE
-            backView.setOnClickListener { requireActivity().finishAfterTransition() }
```
Comment (grote) [inline app/src/main/java/com/stevesoltys/seedvault/ui/recoverycode/RecoveryCodeInputFragment.kt:109]: Why was the back view removed here? Is the back functionality provided in a different way now?

File: app/src/main/java/com/stevesoltys/seedvault/ui/storage/StorageRootsFragment.kt [hunk 7]
```diff
@@ -56,13 +61,17 @@ internal class StorageRootsFragment : Fragment(), StorageRootClickedListener {
...
         warningText = v.findViewById(R.id.warningText)
         divider = v.findViewById(R.id.divider)
         listView = v.findViewById(R.id.listView)
         progressBar = v.findViewById(R.id.progressBar)
-        backView = v.findViewById(R.id.backView)
+        skipView = v.findViewById(R.id.skipView)
+
+        val icon: Drawable = suw_layout.getIcon()
+        icon.setTintList(getColorAccent(requireContext()))
+        suw_layout.setIcon(icon)
```
Comment (grote) [inline app/src/main/java/com/stevesoltys/seedvault/ui/storage/StorageRootsFragment.kt:74]: It is too bad that `suw_layout` can't handle icon tinting internally. Kinda annoying having to set it everywhere.

File: app/src/main/res/layout/fragment_storage_check.xml [hunk 8]
```diff
@@ -1,87 +1,83 @@
...
+            style="?android:progressBarStyleLarge"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="32dp"
+            android:layout_marginEnd="16dp"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/titleView"
```
Comment (grote) [inline app/src/main/res/layout/fragment_storage_check.xml:49]: Looks like the constraints here have changed. Why?
When gradle build is working again, I need to check these layouts again. The diff is hard to follow.

File: Android.bp [hunk 9]
```diff
@@ -32,6 +32,8 @@ android_app {
         "androidx.lifecycle_lifecycle-livedata-ktx",
         "androidx-constraintlayout_constraintlayout",
         "com.google.android.material_material",
+        "setupcompat",
+        "setupdesign",
```
Comment (grote) [inline Android.bp:36]: Are these in AOSP?

File: app/src/main/res/layout/header_snapshots.xml [hunk 10]
```diff
@@ -5,29 +5,11 @@
...
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/imageView" />
+    <com.google.android.setupdesign.GlifLayout
+        android:id="@+id/setup_wizard_layout"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:icon="@drawable/ic_cloud_download"
+        app:sucHeaderText="@string/restore_storage_choose_snapshot" />
 
 </androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (grote) [inline app/src/main/res/layout/header_snapshots.xml:15]: If we proceed with wrapping everything in `GlifLayout`, maybe we don't even need `ConstraintLayout` here anymore?

File: app/src/main/res/values/colors.xml [hunk 11]
```diff
@@ -11,4 +11,6 @@
     <color name="divider">#20ffffff</color>
     <color name="green">#558B2F</color>
     <color name="yellow">#F9A825</color>
+
+    <color name="sud_color_accent_glif_v3_light">@color/accent</color>
```
Comment (grote) [inline app/src/main/res/values/colors.xml:15]: Any reason not to use `@color/accent` right away? Or is this redefining `sud_color_accent_glif_v3_light` from the design library?

**Review by grote — CHANGES_REQUESTED** [review]
Just had a quick look.

Not sure if we *really* need to pull in two new AOSP libraries or if in the long run it would be better to duplicate what they do in our code.

**Comment by grote** [comment]
Should also watch out for stuff like

```
12-30 03:07:50.030 21392 21392 E SetupLibrary: [PartnerCustomizationLayout] Attribute sucUsePartnerResource not found in ComponentInfo{com.stevesoltys.seedvault/com.stevesoltys.seedvault.restore.RestoreActivity}
```

File: Android.bp (hunk 9 above)
Comment (chirayudesai) [reply Android.bp:36]: Yeah, I noted that in the commit message.

https://android.googlesource.com/platform/external/setupcompat/
https://android.googlesource.com/platform/external/setupdesign/

File: app/src/main/res/layout/fragment_storage_check.xml (hunk 8 above)
Comment (chirayudesai) [reply app/src/main/res/layout/fragment_storage_check.xml:49]: There's no `backButton` anymore.

File: app/src/main/res/values/colors.xml (hunk 11 above)
Comment (chirayudesai) [reply app/src/main/res/values/colors.xml:15]: The latter.

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (chirayudesai) [reply app/src/main/res/values/strings.xml:152]: I agree.

File: gradle/dependencies.gradle (hunk 3 above)
Comment (chirayudesai) [reply gradle/dependencies.gradle:27]: I wrote it in the commit message I think, it's a simple:

`m setupcompat setupdesign`

**Review by chirayudesai — COMMENTED** [review]
This is what it looks like now:

<details>
<summary>Click to view screenshots</summary>

Screenshot_20220104-065426
Screenshot_20220104-065429
</details>

**Comment by chirayudesai** [comment]
> This is what it looks like now:
> 
> Click to view screenshots

@grote One thought I had was we could just do this UI ourselves, it looks simple enough - we don't use any complicated stuff available in the `setup*` libraries, and they're OSS Apache2 so we can freely copy-paste things, which at this point might be easier.

**Comment by chirayudesai** [comment]
Leaving this open for now but it won't be merged.

Doing it piecemeal, split out one commit as https://github.com/seedvault-app/seedvault/pull/358

**Comment by chirayudesai** [comment]
Will be included in https://github.com/seedvault-app/seedvault/pull/392
